exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -207,6 +223,7 @@ func (r *AWSManagedMachinePoolReconciler) reconcileNormal(
 
 	ekssvc := eks.NewNodegroupService(machinePoolScope)
 	ec2svc := r.getEC2Service(ec2Scope)
+	asgsvc := r.getASGService(clusterScope)
```

Comment:
Only a feeling, didn't check in detail: shouldn't `ekssvc` have what's needed? We didn't have an `asgsvc` for EKS code before, so why should it be needed now?

Probably related to interfaces again – probably `GetLifecycleHooks(scope scope.LifecycleHookScope)` rather fits into a new interface implemented _by_ the (managed) machine pool scope type?

Reply:
So I tried to keep code duplication to a minimum by having the Lifecycle Hook reconciliation in one place since it should be identical between a ManagedMachinePool and a MachinePool (since both are using ASG in the backend), and therefore this lived in the asgsvc since it wouldn't make any sense to have the ekssvc into the MachinePool controller.
I'll change it to have the same logic in both the EKS NodeGroup Service (which reconciles the ManagedMachinePool) and the ASG Service (which is used to reconcile the MachinePool)

Reply:
It's now all in `pkg/cloud/services/eks/nodegroup.go` and that looks great, thank you!